### PR TITLE
Introduce a migration guide for 3.10 new Zone error

### PR DESCRIPTION
**Pull request by Hixie**
Migration guide for https://github.com/flutter/flutter/pull/122836

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/zone-errors.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+* [`Zone`][]
+* [`BindingBase.debugZoneErrorsAreFatal`][]
+
+Relevant issues:
+
+* [Issue 94123][]: Flutter framework does not warn when ensureInitialized is called in a different zone than runApp
+
+Relevant PRs:
+
+* [PR 122836][https://github.com/flutter/flutter/pull/122836]: Assert that runApp is called in the same zone as binding.ensureInitialized
```
Comment (parlough) [inline src/release/breaking-changes/zone-errors.md:137]: You already defined this link below:

```suggestion
* [PR 122836][]: Assert that runApp is called in the same zone as binding.ensureInitialized
```

File: src/release/breaking-changes/zone-errors.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+https://github.com/flutter/flutter/issues/
```
Comment (parlough) [inline src/release/breaking-changes/zone-errors.md:138]: Seems like this was accidental:

```suggestion
```

File: src/release/breaking-changes/zone-errors.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+It is important to use the same zone when calling `ensureInitialized` on the
+binding as when calling `runApp` later.
+To make this warning fatal, set BindingBase.debugZoneErrorsAreFatal to true before
+the bindings are initialized (i.e. as the first statement in `void main() { }`).
+[...]
+═══════════════════════════════════════════════════════════════════════════════════
+```
+
+The warning can be made fatal by setting [`BindingBase.debugZoneErrorsAreFatal`] to true.
+This flag may be changed to default to true in a future version of Flutter.
```
Comment (parlough) [inline src/release/breaking-changes/zone-errors.md:59]:
```suggestion
This flag might be changed to default to true in a future version of Flutter.
```

https://developers.google.com/style/word-list#may

File: src/release/breaking-changes/zone-errors.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
+---
+title: "Zone mismatch" message
```
Comment (parlough) [inline src/release/breaking-changes/zone-errors.md:2]: To fix the Jekyll error:

```suggestion
title: "\"Zone mismatch\" message"
```

**Review by parlough — COMMENTED** [review]
Thanks for writing this document! The background and migration guide read well and make sense to me.

I noted a few issues. Could you also add a link to this doc from the breaking changes index for 3.10? https://github.com/flutter/website/blob/main/src/release/breaking-changes/index.md?plain=1#L43-L44

File: src/release/breaking-changes/zone-errors.md (hunk 1 above)
Comment (Hixie) [reply src/release/breaking-changes/zone-errors.md:137]: fixed

File: src/release/breaking-changes/zone-errors.md (hunk 2 above)
Comment (Hixie) [reply src/release/breaking-changes/zone-errors.md:138]: fixed

File: src/release/breaking-changes/zone-errors.md (hunk 3 above)
Comment (Hixie) [reply src/release/breaking-changes/zone-errors.md:59]: fixed

File: src/release/breaking-changes/zone-errors.md (hunk 4 above)
Comment (Hixie) [reply src/release/breaking-changes/zone-errors.md]: fixed

**Comment by Hixie** [comment]
> Could you also add a link to this doc from the breaking changes index for 3.10? https://github.com/flutter/website/blob/main/src/release/breaking-changes/index.md?plain=1#L43-L44

Done. Also fixed the missing items that were in that list.

**Comment by parlough** [comment]
Thanks! Those link definitions were mostly there by accident and weren't released yet in 3.10 or were already in the list above. I pushed fixes so that the index now matches the state of things.

**Comment by Hixie** [comment]
Thanks. I got a bit confused with the list there when doing the merge.

**Review by parlough — APPROVED** [review]
Thanks for writing this and fixing those issues. Looks good to me!

**Review by parlough — APPROVED** [review]
Thanks for writing this and fixing those issues. Looks good to me!

**Pull request merged**